nwg-dock crashes if a floating window is split in sway; references compile-time directory

**Describe the bug**
If a floating window is split horizontally or vertically, nwg-dock crashes with the following output:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5ebcb9]

goroutine 35 [running]:
main.createTask(...)
	/home/cari/Downloads/nwg-dock/tools.go:235
main.listTasks()
	/home/cari/Downloads/nwg-dock/tools.go:209 +0xc79
main.getTaskChangesChannel.func2()
	/home/cari/Downloads/nwg-dock/tools.go:98 +0x4c
created by main.getTaskChangesChannel
	/home/cari/Downloads/nwg-dock/tools.go:95 +0x125
```

It's referencing the folder where I compiled the application, which, of course, no longer exists. I'm not sure why it might still be trying to access anything in this folder after compiling and installing.

The same crash occurs when attempting to launch nwg-dock when a split floating window already exists.

**To Reproduce**
1. Launch nwg-dock.
2. Open a floating window in sway.
3. Split the window either horizontally or vertically.
4. nwg-dock crashes.

**Expected behavior**
nwg-dock should not crash, nor should it attempt to access the compile-time directory after being compiled and installed.

**Desktop (please complete the following information):**
 - Fedora Linux 38
 - go 1.20.3
 - gtk-layer-shell 0.8.1-1

Thanks for taking a look! ☺️

--- a/main.go
+++ b/main.go
@@ -140,8 +140,7 @@ func buildMainBox(tasks []task, vbox *gtk.Box) {
 
 	alreadyAdded = nil
 	for _, task := range tasks {
-		// nwggrid is the default launcher, we don't want to see it as a task
-		if !inPinned(task.ID) && task.ID != "nwggrid" {
+		if !inPinned(task.ID) {
 			instances := taskInstances(task.ID, tasks)
 			if len(instances) == 1 {
 				button := taskButton(task, instances)
@@ -158,10 +157,13 @@ func buildMainBox(tasks []task, vbox *gtk.Box) {
 	}
 
 	if !*noWs {
-		wsButton, _ := gtk.ButtonNew()
-		wsPixbuf, err := gdk.PixbufNewFromFileAtSize(filepath.Join(dataHome, fmt.Sprintf("nwg-dock/images/%v.svg", currentWsNum)),
+		wsButton, e := gtk.ButtonNew()
+		if e != nil {
+			return
+		}
+		wsPixbuf, e := gdk.PixbufNewFromFileAtSize(filepath.Join(dataHome, fmt.Sprintf("nwg-dock/images/%v.svg", currentWsNum)),
 			imgSizeScaled, imgSizeScaled)
-		if err == nil {
+		if e == nil {
 			wsImage, _ := gtk.ImageNewFromPixbuf(wsPixbuf)
 			wsButton.SetImage(wsImage)
 			wsButton.SetAlwaysShowImage(true)
@@ -174,11 +176,11 @@ func buildMainBox(tasks []task, vbox *gtk.Box) {
 					targetWsNum = activeWorkspace
 
 					glib.TimeoutAdd(0, func() bool {
-						wsPixbuf, err := gdk.PixbufNewFromFileAtSize(filepath.Join(dataHome, fmt.Sprintf("nwg-dock/images/%v.svg", activeWorkspace)),
+						wsPixbuf, e = gdk.PixbufNewFromFileAtSize(filepath.Join(dataHome, fmt.Sprintf("nwg-dock/images/%v.svg", activeWorkspace)),
 							imgSizeScaled, imgSizeScaled)
 
-						if err == nil {
-							wsImage, _ := gtk.ImageNewFromPixbuf(wsPixbuf)
+						if e == nil {
+							wsImage, _ = gtk.ImageNewFromPixbuf(wsPixbuf)
 							wsButton.SetImage(wsImage)
 						} else {
 							log.Warnf("Unable set set workspace image: %v", activeWorkspace)
@@ -219,7 +221,9 @@ func buildMainBox(tasks []task, vbox *gtk.Box) {
 				return false
 			})
 		}
-		mainBox.PackStart(wsButton, false, false, 0)
+		if wsButton != nil {
+			mainBox.PackStart(wsButton, false, false, 0)
+		}
 	}
 
 	if !*noLauncher && *launcherCmd != "" {

--- a/main.go
+++ b/main.go
@@ -22,7 +22,7 @@ import (
 	"github.com/gotk3/gotk3/gtk"
 )
 
-const version = "0.3.3"
+const version = "0.3.4"
 
 type WindowState int
 

--- a/tools.go
+++ b/tools.go
@@ -413,10 +413,8 @@ func taskButton(t task, instances []task) *gtk.Box {
 func taskMenu(taskID string, instances []task) gtk.Menu {
 	menu, _ := gtk.MenuNew()
 
-	iconName, err := getIcon(taskID)
-	if err != nil {
-		log.Warn(err)
-	}
+	iconName, _ := getIcon(taskID)
+
 	for _, instance := range instances {
 		menuItem, _ := gtk.MenuItemNew()
 		hbox, _ := gtk.BoxNew(gtk.ORIENTATION_HORIZONTAL, 6)
@@ -729,7 +727,7 @@ func getIcon(appName string) (string, error) {
 			}
 		}
 	}
-	return "", errors.New("couldn't find the icon")
+	return "", errors.New(fmt.Sprintf("couldn't find the icon for %s", appName))
 }
 
 func searchDesktopDirs(badAppID string) string {

--- a/tools.go
+++ b/tools.go
@@ -201,14 +201,23 @@ func listTasks() ([]task, error) {
 
 		// create tasks from cons which represent tasks
 		for _, con := range descendants {
-			tasks = append(tasks, createTask(con, wsNum))
+			t, err := createTask(con, wsNum)
+			if err == nil {
+				tasks = append(tasks, *t)
+			} else {
+				log.Warn(err)
+			}
 		}
 
 		fNodes := w.FloatingNodes
 		for _, con := range fNodes {
-			tasks = append(tasks, createTask(*con, wsNum))
+			t, err := createTask(*con, wsNum)
+			if err == nil {
+				tasks = append(tasks, *t)
+			} else {
+				log.Warn(err)
+			}
 		}
-
 	}
 	sort.Slice(tasks, func(i int, j int) bool {
 		return tasks[i].WsNum < tasks[j].WsNum
@@ -226,20 +235,27 @@ func findDescendants(con sway.Node) {
 	}
 }
 
-func createTask(con sway.Node, wsNum int64) task {
-	t := task{}
+func createTask(con sway.Node, wsNum int64) (*task, error) {
+	t := &task{}
 	t.conID = con.ID
 	if con.AppID != nil {
 		t.ID = *con.AppID
-	} else {
+	} else if con.WindowProperties != nil {
 		wp := *con.WindowProperties
 		t.ID = wp.Class
+	} else {
+		return nil, errors.New("damaged Node data received, task skipped")
 	}
 	t.Name = con.Name
-	t.PID = *con.PID
+	if con.PID != nil {
+		t.PID = *con.PID
+	} else {
+		return nil, errors.New("damaged Node data received, task skipped")
+	}
+
 	t.WsNum = wsNum
 
-	return t
+	return t, nil
 }
 
 func workspaceNum(workspaces []sway.Workspace, name string) int64 {

--- a/tools.go
+++ b/tools.go
@@ -733,15 +733,28 @@ func getIcon(appName string) (string, error) {
 }
 
 func searchDesktopDirs(badAppID string) string {
-	b4Hyphen := strings.Split(badAppID, "-")[0]
+	b4Separator := strings.Split(badAppID, "-")[0]
 	for _, d := range appDirs {
 		items, _ := os.ReadDir(d)
 		for _, item := range items {
-			if strings.Contains(item.Name(), b4Hyphen) {
+			if strings.Contains(item.Name(), b4Separator) {
 				//Let's check items starting from 'org.' first
-				if strings.Count(item.Name(), ".") > 1 {
+				if strings.Count(item.Name(), ".") > 1 && strings.HasSuffix(item.Name(),
+					fmt.Sprintf("%s.desktop", badAppID)) {
 					return filepath.Join(d, item.Name())
+				} else {
+					return ""
 				}
+			}
+		}
+	}
+	// exceptions like "class": "VirtualBox Manager" & virtualbox.desktop
+	b4Separator = strings.Split(badAppID, " ")[0]
+	for _, d := range appDirs {
+		items, _ := os.ReadDir(d)
+		for _, item := range items {
+			if strings.Contains(strings.ToUpper(item.Name()), strings.ToUpper(b4Separator)) {
+				return filepath.Join(d, item.Name())
 			}
 		}
 	}

